client/my-sites/stats/stats-notices/free-plan-purchase-success-notice.tsx

```diff
@@ -32,14 +33,19 @@ const handleUpgradeClick = (
 
 const FreePlanPurchaseSuccessJetpackStatsNotice = ( {
 	siteId,
-	onNoticeViewed,
+	isOdysseyStats,
 }: StatsNoticeProps ) => {
 	const translate = useTranslate();
-	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 	const [ noticeDismissed, setNoticeDismissed ] = useState( false );
+	const [ paramRemoved, setParamRemoved ] = useState( false );
 
 	useEffect( () => {
-		onNoticeViewed && onNoticeViewed();
+		if ( paramRemoved ) {
+			return;
+		}
+		// Ensure it runs only once.
+		setParamRemoved( true );
+		removeStatsPurchaseSuccessParamFromCurrentUrl( isOdysseyStats );
 	} );
```

Comment:
```suggestion
	}, [ paramRemoved, isOdysseyStats ] );
```

<img width="1075" alt="截圖 2023-08-24 上午1 47 03" src="https://github.com/Automattic/wp-calypso/assets/6869813/30b89087-ce99-4093-ae55-83109614dfdd">

Reply:
Fixed in 19852b4719